pkg/fanal/handler/handler.go

```diff
@@ -75,5 +77,33 @@ func (m Manager) PostHandle(ctx context.Context, result *analyzer.AnalysisResult
 			return xerrors.Errorf("post handler error: %w", err)
 		}
 	}
+
+	// Overwrite package identifier info for system packages
+	if result != nil {
+		result.PackageInfos = overwritePackageIdentifiers(result.PackageInfos, result.OS)
+	}
+	if blob != nil {
+		blob.PackageInfos = overwritePackageIdentifiers(blob.PackageInfos, blob.OS)
+	}
```

Comment:
`result` and `blob` are pointers.
But if i remember corerctly - we only need to change `blob`
https://github.com/aquasecurity/trivy/blob/ad977a4256920d6269e84c82e4f42bbe244a2c5a/pkg/fanal/handler/sysfile/filter.go#L99-L102

Reply:
done at https://github.com/aquasecurity/trivy/pull/5439/commits/9a162b2d6031e362fad8a5b4ba493bbb65706984